.github/workflows/docs-check.yml

```diff
@@ -38,14 +38,13 @@ jobs:
       - name: Check wordlist is sorted
         run: scripts/wordlist_sort_check.sh
 
+      - name: Spellcheck
```

Comment:
Moved before the CLI reference generation as otherwise `spellcheck` fails on all custom headers ids.
We use custom header ids for subcommand arguments (`-` instead of `--`) so that previous links are not broken.

CLI docs are already checked as they are part of the source code.

Reply:
The solution probably is to switch to `mdx` and explicitly export TOC metadata via the `toc` global. See https://github.com/facebook/docusaurus/issues/6201#issuecomment-2012134123

Reply:
so before did it pass validation because we used backticks `` for those values, and can't we use this approach as well or does it not format as we want it to?

Reply:
Unfortunately spellcheck is also applied to the custom id part, and this part can't be escaped.
e.g. ```### `--myArg` {#-myArg}```

Reply:
Fixed it by programmatically generating the TOC. This open the door for more customizations down the line.

<img width="339" alt="Capture d’écran 2024-03-25 à 10 19 04" src="https://github.com/ChainSafe/lodestar/assets/359723/0b19b560-9bef-430b-b12e-846f103ab0ad">

Reply:
Do you have these changes deployed somewhere to review? I guess I can run the branch locally as well to review the rendered UI. This is similar to https://github.com/ethereum/beacon-APIs/issues/415 would be great to have but likely a bit of effort.

Reply:
But since we plan to add more complex / interactive code examples in the docs, it might be warranted to implement as it makes reviewing so much easier

Reply:
You can see it deployed here: https://jeluard.github.io/lodestar/validator-management/validator-cli